test.js

```diff
@@ -3,20 +3,80 @@
 const assert = require("assert");
 const process = require("process");
 const execa = require("execa");
+const createTestServer = require("create-test-server");
```

Comment:
Can you use [nock](https://github.com/nock/nock#usage) instead? It should work transparently and we don't have to redefine `registry` that way.

Reply:
I tried to use `nock`, but failed, I think it's because we are using `execa`, nock only override `http.request` of main thread.

Any suggestion?

Reply:
`nock` only works in-process (some discussion about this in https://github.com/nock/nock/issues/1911). I will set up a local HTTP server for the registry mocking instead which can be passed via `--registry` during the tests.